physics/ground_to_ground_projectile_motion.cpp

```diff
@@ -0,0 +1,133 @@
+    double Viy = initial_velocity * (std::sin(degrees_to_radians(angle))); // calculate y component of the initial velocity
+    return (std::pow(Viy, 2) / (2.0 * gravity));
+}
+}  // namespace physics
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test() {
+    // initial input variables
+    double initial_velocity = 5.0;  // double initial_velocity input
+    double angle = 40.0;            // double angle input
+
+    // 1st test
+    double expected_time_of_flight = 0.655;  // expected time output
+    double flight_time_output =
+        std::round(physics::time_of_flight(initial_velocity, angle) * 1000.0) /
+        1000.0;  // round output to 3 decimal places
+
+    std::cout << "Projectile Flight Time (double)" << std::endl;
+    std::cout << "Input Initial Velocity: " << initial_velocity << std::endl;
+    std::cout << "Input Angle: " << angle << std::endl;
+    std::cout << "Expected Output: " << expected_time_of_flight << std::endl;
+    std::cout << "Output: " << flight_time_output << std::endl;
+    assert(flight_time_output == expected_time_of_flight);
+    std::cout << "TEST PASSED" << std::endl << std::endl;
+
+    // 2nd test
+    double expected_horizontal_range = 2.51; // expected range output
+    double horizontal_range_output =
+        std::round(physics::horizontal_range(initial_velocity, angle,
+                                             flight_time_output) *
+                   100.0) /
+        100.0;  // round output to 2 decimal places
+
+    std::cout << "Projectile Horizontal Range (double)" << std::endl;
+    std::cout << "Input Initial Velocity: " << initial_velocity << std::endl;
+    std::cout << "Input Angle: " << angle << std::endl;
+    std::cout << "Input Time Of Flight: " << flight_time_output << std::endl;
+    std::cout << "Expected Output: " << expected_horizontal_range << std::endl;
+    std::cout << "Output: " << horizontal_range_output << std::endl;
+    assert(horizontal_range_output == expected_horizontal_range);
+    std::cout << "TEST PASSED" << std::endl << std::endl;
+
+    // 3rd test
+    double expected_max_height = 0.526; // expected height output
+    double max_height_output =
+        std::round(physics::max_height(initial_velocity, angle) * 1000.0) /
+        1000.0;  // round output to 3 decimal places
+
+    std::cout << "Projectile Max Height (double)" << std::endl;
+    std::cout << "Input Initial Velocity: " << initial_velocity << std::endl;
+    std::cout << "Input Angle: " << angle << std::endl;
+    std::cout << "Expected Output: " << expected_max_height << std::endl;
+    std::cout << "Output: " << max_height_output << std::endl;
+    assert(max_height_output == expected_max_height);
+    std::cout << "TEST PASSED" << std::endl << std::endl;
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();  // run self-test implementations
+    return 0;
+}
```

Comment:
Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```